base/markdown/Common/block.jl

```diff
@@ -198,7 +198,7 @@ function list(stream::IO, block::MD)
                 c = read(stream, Char)
                 if c == '\n'
                     eof(stream) && break
-                    next = peek(stream)
+                    next = Char(peek(stream)) # ok since we only compare with ASCII
                     if next == '\n'
```

Comment:
Wouldn't it be better to leave `next` as a byte here and compare with `UInt8('\n')`?